src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -323,7 +323,7 @@ void GetInputTabularMonthly(EnergyPlusData &state)
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     auto &ort = state.dataOutRptTab;
 
-    if (!(state.files.outputControl.tabular || state.files.outputControl.sqlite)) {
+    if (!state.files.outputControl.writeTabular(state)) {
```

Comment:
I think I'd like to take a crack at this where `state.dataOutRptTab->WriteTabularFiles` is set once very early in the simulation and we don't need to keep calling this function.  As soon as we have read inputs, we should be able to assign that flag.  

I like that the function collects all the flags in one place -- that's always welcomed.  The flag checks themselves may need a little extra work as described above, but regardless it's a good step.

Reply:
Sounds good.